corehq/messaging/scheduling/tasks.py

```diff
@@ -531,3 +537,34 @@ def handle_case_timed_schedule_instance(case_id, schedule_instance_id, domain):
 def delete_schedule_instances_for_cases(domain, case_ids):
     for case_id in case_ids:
         delete_schedule_instances_by_case_id(domain, case_id)
+
+
+@periodic_task(run_every=crontab(minute='0', hour='1'), queue=settings.CELERY_PERIODIC_QUEUE)
+def delete_unused_messaging_images():
```

Comment:
Can you talk more about the decision to do this as a periodic task rather than say something that gets called when an HTML email is saved?

Reply:
Doing this on every email save seemed overkill to me, and I was worried that if there end up being many emails in the system that this would become a longer and longer running task. The downside is that there might be unused images stored for up to a day in the blob db.
However, since we limit the upload size to 1MB per email, I think it would be hard to 'abuse' this feature, as you'd have to upload many images for this to cause storage issues on our side.

Reply:
I was thinking more about scoping this to one email, and then operating on that emails blobs when the email is saved or deleted.  I don't think there's an easy way to do that at present, looking at how the images are stored.  `EmailImage` uses `domain` as `parent_id`, but `BlobMeta` supports both `domain` and `parent_id` as independent fields, usually using the latter to point to the object that conceptually owns the blob (an email config, in this case).  For instance, here's where blobs are saved in `BlobMixin`:
https://github.com/dimagi/commcare-hq/blob/3e11c9ebf06cbe03a69cdd3391bf7a64a9a8aa3d/corehq/blobs/mixin.py#L150-L157
With that connection it'd be easy to fetch all blobs ostensibly associated with a particular email config and compare it against the current version of the email.

Reply:
Yes, that was an option I explored. However, you'd still need a way of deleting images that were added to emails that were never saved.